include/seqan/graph_algorithms/depth_first_search.h

```diff
@@ -121,28 +123,72 @@ void depthFirstSearch(TPredecessorMap & predecessor,
     typedef typename Iterator<TGraph, VertexIterator>::Type TVertexIterator;
     typedef typename VertexDescriptor<TGraph>::Type TVertexDescriptor;
     typedef typename Value<TPredecessorMap>::Type TPredVal;
+    typedef typename Iterator<Graph<TSpec>, AdjacencyIterator>::Type TAdjacencyIterator;
+    enum Task {EXPLORE, FINISH};
+    typedef std::pair<TVertexDescriptor, Task> StackItem;
```

Comment:
You can remove this typedef and declare the vector directly with this type. Then replace all expressions of the form:
```cpp
StackItem var = ...
```
with ```auto``` and use ```emplace_back``` to push values to the vector.